File: backend/app/models/notification.py
from __future__ import annotations

import enum
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, Integer, String, Text, Boolean
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base
from typing import Optional


class NotificationType(str, enum.Enum):
    review_submit = "review_submit"  # 提交审核
    review_approved = "review_approved"  # 审核通过
    review_rejected = "review_rejected"  # 审核驳回


class Notification(Base):
    __tablename__ = "notifications"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"), nullable=False)
    type: Mapped[NotificationType] = mapped_column(String(30), nullable=False)
    title: Mapped[str] = mapped_column(String(200), nullable=False)
    content: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    related_id: Mapped[Optional[int]] = mapped_column(Integer, nullable=True)
    is_read: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    created_at: Mapped[datetime] = mapped_column(
        DateTime, default=datetime.now, nullable=False
    )

File: backend-go/models/notification.go
package models

import (
	"time"
)

const (
	NotificationTypeReviewSubmit           = "review_submit"
	NotificationTypeReviewApproved         = "review_approved"
	NotificationTypeReviewRejected         = "review_rejected"
	NotificationTypeRoleUpdated            = "role_updated"
	NotificationTypeRolePermissionsUpdated = "role_permissions_updated"

	NotificationTypeTaskCreated         = "inspection_task_created"
	NotificationTypeTaskSubmitted       = "inspection_task_submitted"
	NotificationTypeTaskRecheckPassed   = "inspection_task_recheck_passed"
	NotificationTypeTaskRecheckFailed   = "inspection_task_recheck_failed"
	NotificationTypeTaskManualReview    = "inspection_task_manual_review"
	NotificationTypeTaskConfirmed       = "inspection_task_confirmed"
	NotificationTypeTaskRejected        = "inspection_task_rejected"
)

const (
	NotificationChannelInternal = "internal"
)

type Notification struct {
	ID        string    `orm:"column(id);pk;size(36)" json:"id"`
	UserID    string    `orm:"column(user_id);size(36);index" json:"user_id"`
	Type      string    `orm:"column(type);size(30)" json:"type"`
	Title     string    `orm:"column(title);size(200)" json:"title"`
	Content   string    `orm:"column(content);type(text);null" json:"content"`
	RelatedID string    `orm:"column(related_id);size(36);null" json:"related_id"`
	Channel   string    `orm:"column(channel);size(20);default(internal)" json:"channel"`
	IsRead    bool      `orm:"column(is_read);default(false)" json:"is_read"`
	CreatedAt time.Time `orm:"column(created_at);auto_now_add;type(datetime)" json:"created_at"`
}

func (n *Notification) TableName() string {
	return "notifications"
}

type NotificationDict struct {
	ID        string    `orm:"column(id);pk;size(36)" json:"id"`
	Category  string    `orm:"column(category);size(20);index" json:"category"`
	GroupKey  string    `orm:"column(group_key);size(100);index" json:"group_key"`
	DictKey   string    `orm:"column(dict_key);size(200)" json:"dict_key"`
	DictValue string    `orm:"column(dict_value);size(200)" json:"dict_value"`
	IsActive  bool      `orm:"column(is_active);default(true)" json:"is_active"`
	IsPrivate bool      `orm:"column(is_private);default(false)" json:"is_private"`
	SortOrder int       `orm:"column(sort_order);default(0)" json:"sort_order"`
	CreatedAt time.Time `orm:"column(created_at);auto_now_add;type(datetime)" json:"created_at"`
	UpdatedAt time.Time `orm:"column(updated_at);auto_now;type(datetime)" json:"updated_at"`
}

func (n *NotificationDict) TableName() string {
	return "notification_dict"
}
